xdsl/utils/comparisons.py

```diff
@@ -32,31 +32,59 @@
 their bit representations are the same.
 """
 
-from xdsl.dialects.builtin import Signedness
-
 
 def unsigned_upper_bound(bitwidth: int) -> int:
     """
     The maximum representable value + 1.
     """
-    _, ub = Signedness.UNSIGNED.value_range(bitwidth)
-    return ub
+    return 1 << bitwidth
 
 
 def signed_lower_bound(bitwidth: int) -> int:
     """
     The minimum representable value.
     """
-    lb, _ = Signedness.SIGNED.value_range(bitwidth)
-    return lb
+    return -((1 << bitwidth) >> 1)
 
 
 def signed_upper_bound(bitwidth: int) -> int:
     """
     The maximum representable value + 1.
     """
-    _, ub = Signedness.SIGNED.value_range(bitwidth)
-    return ub
+    return 1 << (bitwidth - 1)
+
+
+def unsigned_value_range(bitwidth: int) -> tuple[int, int]:
+    """
+    For a given bitwidth, returns (min, max+1), where min and max are the smallest and
```

Comment:
why not just `(min, max)`?

Reply:
It slightly complicates some calculations, which is not a big deal, but a little annoying. I also find that this inclusive-lb-exclusive-ub form is more intuitive.

Reply:
I'm not sure if this is a mathematician thing but this can also be written by saying that the integer takes values in the range `[min, max)`

Reply:
this is a me being ambiguous thing, I meant for the `()` to be tuple begin end markers, I'll rephrase

Reply:
How's the new version?